backend/fms_core/migrations/0065_v4_12_0.py

```diff
@@ -0,0 +1,66 @@
+
+ADMIN_USERNAME = 'biobankadmin'
+
+# adding the following steps
+# important !! the column name will change because it doesn't exist in SAMPLE_QC_TEMPLATE
+# column sample kind to be added to template
+def add_sample_qc_distinction_dna_rna(apps, schema_editor):
+    Step = apps.get_model("fms_core", "Step")
+    Protocol = apps.get_model("fms_core", "Protocol")
+    StepSpecification = apps.get_model("fms_core", "StepSpecification")
+
+    STEPS = [
+        {"name": "Sample QC (DNA)", "protocol_name": "Sample Quality Control","expected_sample_type": "EXTRACTED_SAMPLE", "specifications": [
+            {"display_name": "Sample QC Type", "sheet_name": "SampleQC", "column_name": "Sample Kind", "value": "DNA"}]
+        },
+        {"name": "Sample QC (RNA)", "protocol_name": "Sample Quality Control","expected_sample_type": "EXTRACTED_SAMPLE", "specifications": [
+            {"display_name": "Sample QC Type", "sheet_name": "SampleQC", "column_name": "Sample Kind", "value": "RNA"}]
+        }
+    ]
+
+    with reversion.create_revision(manage_manually=True):
+        admin_user = User.objects.get(username=ADMIN_USERNAME)
+        reversion.set_comment(f"Create the basic initial workflows.")
+        reversion.set_user(admin_user)
+        #  existing step sample QC to be removed?
+        for step_info in STEPS:
+            protocol = Protocol.objects.get(name=step_info["protocol_name"])
+            step = Step.objects.create(name=step_info["name"],
+                                       protocol_id=protocol.id,
+                                       type="PROTOCOL",
+                                       needs_placement=False,
+                                       needs_planning=False,
+                                       expected_sample_type=step_info["expected_sample_type"],
+                                       created_by_id=admin_user.id,
+                                       updated_by_id=admin_user.id)
+            reversion.add_to_revision(step)
+            for specification in step_info["specifications"]:
+                step_specification = StepSpecification.objects.create(name=specification["display_name"],
+                                                                      sheet_name=specification["sheet_name"],
+                                                                      column_name=specification["column_name"],
+                                                                      value=specification["value"],
+                                                                      step=step,
+                                                                      created_by_id=admin_user.id,
+                                                                      updated_by_id=admin_user.id)
+                reversion.add_to_revision(step_specification)
+        step = Step.objects.get(name="Sample QC")
+        dnaStep = Step.objects.get(name="Sample QC (DNA)")
+        if step:
```

Comment:
There is an issue with the logic in this section ... You only use the RNA step in SampleNextStep and StepOrder. Those step are condition to the previous extraction step used previously in the workflows. so for each instance of sample next step you need to check in the list of linked studies the workflow step for extraction and verify if it is DNA or RNA. If there is a conflict between the studies you would need to create a new sampleNextStep and SampleNextStepByStudy for that other kind (i don't think there is any in the system so far but we cannot assume. Same goes for StepOrder except that in that case you can't have conflict you have just one workflow by StepOrder. You just need to use the same kind for the step connected as the one for the extraction step.